uiServices/src/xmlWriter.go

```diff
@@ -53,3 +53,24 @@ func testSuiteWriterXml(testSuite testStrategies.TestSuite, configPathDir string
 	}
 	return true
 }
+
+func testCaseWriterXml(testSuite testStrategies.TestDefinition, path string) bool {
+	filename := fmt.Sprintf("%s", path)
+
+	testCaseAsXml, err := xml.MarshalIndent(testSuite, "  ", "    ")
+	if err != nil {
+		log.Error("Failed to marshal to XML. Error:", err)
+		return false
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		log.Error("Failed to create output file. Error:", err)
+		return false
+	}
+	if file != nil {
+		defer file.Close()
+		file.Write(testCaseAsXml)
+	}
+	return true
```

Comment:
if file is nil, you're returning true.

switch it around
```
if file == nil {
     log.Error("File was not created successfully - was nil")
     return false
}
defer file.Close()
file.Write(testCaseAsXml)
return true
```

and you should check that file.Write doesn't return an error either.